### DISCUSSION TITLE
История прослушанных треков за определенный временнОй интервал

### DISCUSSION THREAD

**Discussion by ilyanikolaich** [Chimildic/goofy#51] [Помощь с алгоритмом] [answered]
Не всегда удобно оперировать полученной историей прослушанных треков, исходя только из их количества. Например, как раз для скрипта "очистки" недавно слушанного плейлиста - чтобы вернуться к его непрослушанным трекам, нет необходимости брать историю за 2000 треков - просто потому, что это по времени может быть достаточно давно, и в то время может быть я и слушал какие-то треки из этого плейлиста, которые вот сейчас я бы хотел заново прослушать.
В этой ситуации было бы удобно в скрипте на "очистку" плейлиста от недавно прослушанных треков указать ссылку на сам плейлист и _количество дней  истории, от которой его нужно очистить_.
Например, для вчера слушанного плейлиста будет актуально указать 1 день истории для очистки. А вот например для плейлиста с прошлой недели - уже актуально очистить 7 дней истории, но не больше и не меньше.
Так вот. Если воспользоваться вот таким вот элементарным решением:
```javascript
function RecentTracksForDays( Days ){    
  updateRecentTracks_();  
  let History = RecentTracks.get();
  Filter.rangeDateRel( History, Days, 0 );
  return (History);
};

let FromYesterdayHistory = RecentTracksForDays(1);
// FromYesterdayHistory - история со вчера

let LastWeekHistory = RecentTracksForDays(7);
// FromYesterdayHistory - история за неделю
```
то вроде бы мы и должны получать историю в нужном нам количестве в удобном формате вызова функции.
Недлинное тестирование показало, что вроде как работает похоже на правду.
Существуют ли в такой связке какие-то подводные камни, может быть есть какие-то ограничения или особенности применения?
Спасибо!

**Comment by efremych** [comment, marked as answer]
`let History = RecentTracks.get();` даст 200 треков. _(поправка, оказывается уже нет :) все треки дает)_
И непонятно зачем `updateRecentTracks_();`

А так, лично я никаких подводных камней не обнаруживал.
Для разгребания складов DiscoveryWeekly и т.п. использую недельную историю, жалоб не было

```js
  let recentTracks = RecentTracks.get(1000);
    Filter.rangeDateRel(recentTracks, 7);
```

**Reply by ilyanikolaich** [reply]
> И непонятно зачем `updateRecentTracks_();`

Это для того, чтобы только что проигранные треки зафиксировались в истории, так как по умолчанию `updateRecentTracks_();` отрабатывается раз в 15 минут, и чаще всего последние прослушанные 1-5 треков могут в историю и не попасть перед обработкой. Это актуально, когда музыка нон-стоп слушается.
